File: sdk/runanywhere-swift/Sources/RunAnywhere/Capabilities/SpeakerDiarization/Models/Output/SpeakerDiarizationProfile.swift
//
//  SpeakerDiarizationProfile.swift
//  RunAnywhere SDK
//
//  Speaker profile with statistics
//

import Foundation

// MARK: - Speaker Profile

/// Profile of a speaker with statistics
public struct SpeakerDiarizationProfile: Sendable {

    /// Unique identifier for the speaker
    public let id: String

    /// Speaker embedding vector
    public let embedding: [Float]?

    /// Total speaking time across all segments
    public let totalSpeakingTime: TimeInterval

    /// Number of segments for this speaker
    public let segmentCount: Int

    /// Optional display name
    public let name: String?

    // MARK: - Initialization

    public init(
        id: String,
        embedding: [Float]? = nil,
        totalSpeakingTime: TimeInterval,
        segmentCount: Int,
        name: String? = nil
    ) {
        self.id = id
        self.embedding = embedding
        self.totalSpeakingTime = totalSpeakingTime
        self.segmentCount = segmentCount
        self.name = name
    }
}

File: sdk/runanywhere-flutter/packages/runanywhere/lib/features/speaker_diarization/models/speaker_diarization_profile.dart
/// Profile of a speaker with statistics
/// Matches iOS SpeakerDiarizationProfile from Features/SpeakerDiarization/Models/SpeakerDiarizationProfile.swift
class SpeakerDiarizationProfile {
  /// Unique identifier for the speaker
  final String id;

  /// Speaker embedding vector
  final List<double>? embedding;

  /// Total speaking time across all segments
  final double totalSpeakingTime;

  /// Number of segments for this speaker
  final int segmentCount;

  /// Optional display name
  final String? name;

  const SpeakerDiarizationProfile({
    required this.id,
    this.embedding,
    required this.totalSpeakingTime,
    required this.segmentCount,
    this.name,
  });

  /// Create from map
  factory SpeakerDiarizationProfile.fromJson(Map<String, dynamic> json) {
    return SpeakerDiarizationProfile(
      id: json['id'] as String,
      embedding: (json['embedding'] as List?)?.cast<double>(),
      totalSpeakingTime: (json['totalSpeakingTime'] as num).toDouble(),
      segmentCount: json['segmentCount'] as int,
      name: json['name'] as String?,
    );
  }

  /// Convert to map
  Map<String, dynamic> toJson() => {
        'id': id,
        if (embedding != null) 'embedding': embedding,
        'totalSpeakingTime': totalSpeakingTime,
        'segmentCount': segmentCount,
        if (name != null) 'name': name,
      };
}
